script/run.sh

```diff
@@ -113,29 +117,35 @@ then
         then
             export CMAKE_CXX_FLAGS=
         fi
-        CMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -stdlib=libc++"
 
-        if [ -z "${CMAKE_EXE_LINKER_FLAGS+x}" ]
+        if [ "${CXX}" == "clang++" ]
         then
-            export CMAKE_EXE_LINKER_FLAGS=
+            CMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -stdlib=libc++"
+        #elif [ "${CXX}" == "g++" ]
+        #then
+        #    CMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -nostdinc++ -nodefaultlibs -lc++ -lc++abi -lm -lc -lgcc_s -lgcc -I<libcxx-install-prefix>/include/c++/v1"
         fi
-        CMAKE_EXE_LINKER_FLAGS="${CMAKE_EXE_LINKER_FLAGS} -lc++ -lc++abi"
     fi
 
     which "${CXX}"
     ${CXX} -v
+fi
+
+CMAKE_CUDA_FLAGS=
+if [ ! -z "${CMAKE_CXX_FLAGS+x}" ]
+then
+    CMAKE_CUDA_FLAGS=${CMAKE_CXX_FLAGS}
+fi
```

Comment:
That seems weird to me.

Reply:
Our build scripts provide some `CMAKE_CXX_FLAGS` from the outside because they are applied to all the code that is built by CMake later. This is not the case with the builtin CUDA support because this will not use `CMAKE_CXX_FLAGS` but `CMAKE_CUDA_FLAGS`.
As I did not want to make the CI scripts adapt to the compiler, I simply fill both of the variables with the same content.

Reply:
I am not sure if this is also part of the problem described below (CXX flags that end up on nvcc's command line). You should make sure that nvcc does not get any MSVC CXX flags.

Reply:
I solved both issues by using generator expressions.